datalad/cmd.py

```diff
@@ -194,30 +193,39 @@ def run(self, cmd, log_stdout=True, log_stderr=True, log_online=False,
 
         self.log("Running: %s" % (cmd,))
 
-        if not self.dry:
+        if self.protocol.do_execute_ext_commands:
 
             if shell is None:
                 shell = isinstance(cmd, basestring)
 
-            if self.protocol:
-                self.protocol.write_section(
-                    shlex.split(cmd) if isinstance(cmd, basestring) else cmd)
+            if self.protocol.records_ext_commands:
+                prot_exc = None
+                prot_id = self.protocol.start_section(shlex.split(cmd)
+                                                      if isinstance(cmd,
+                                                                    basestring)
+                                                      else cmd)
+
             try:
                 proc = subprocess.Popen(cmd, stdout=outputstream,
                                         stderr=errstream,
                                         shell=shell,
                                         cwd=cwd or self.cwd,
                                         env=env or self.env)
+                # shell=True allows for piping, multiple commands, etc.,
```

Comment:
Good point. Originally this was a reminder to rethink whether we need the shell-option at all.